core/src/main/java/com/linecorp/armeria/client/retry/RetryRequestStrategy.java

```diff
@@ -0,0 +1,27 @@
+package com.linecorp.armeria.client.retry;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Provides custom condition on whether a failed request should be retried.
+ * @param <I> the request type
+ * @param <O> the response type
+ */
+@FunctionalInterface
+public interface RetryRequestStrategy<I extends Request, O extends Response> {
+
+    static <I extends Request, O extends Response> RetryRequestStrategy<I, O> always() {
```

Comment:
IIUC, this will make a call is repeated infinitely (unless `withMaxAttempts` is specified), because it will retry even when the call was successful.